src/internal.c

```diff
@@ -5455,24 +5467,49 @@ static void BuildMD5(WOLFSSL* ssl, Hashes* hashes, const byte* sender)
     /* make md5 inner */
     md5[0] = ssl->hsHashes->hashMd5 ; /* Save current position */
 
-    wc_Md5Update(&ssl->hsHashes->hashMd5, sender, SIZEOF_SENDER);
-    wc_Md5Update(&ssl->hsHashes->hashMd5, ssl->arrays->masterSecret,SECRET_LEN);
-    wc_Md5Update(&ssl->hsHashes->hashMd5, PAD1, PAD_MD5);
+    ret = wc_Md5Update(&ssl->hsHashes->hashMd5, sender, SIZEOF_SENDER);
+    if (ret != 0) {
+        return ret;
```

Comment:
This is a memory leak in SMALL_STACK.  Please review / fix other spots for resource leaks.